### PR TITLE
Fix some data races

### PR DESCRIPTION
**Pull request by cpuguy83**
After running the test suite with the race detector enabled I found
these gems that need to be fixed.
This is just round one, sadly lost my test results after I built the
binary to test this... (whoops)

### CODE REVIEW

File: api/types/network/network.go [hunk 1]
```diff
@@ -52,6 +52,19 @@ type EndpointSettings struct {
 	MacAddress          string
 }
 
+// Copy makes a deep copy of `EndpointSettings`
+func (es *EndpointSettings) Copy() *EndpointSettings {
+	epCopy := *es
+
```
Comment (AkihiroSuda) [inline api/types/network/network.go]: `es.IPAMConfig` needs to be copied as well?

**Comment by LK4D4** [comment]
Looks like daemon on windows panicked.

**Comment by cpuguy83** [comment]
Suspect I fixed what caused the panic, we'll see.

File: api/types/network/network.go [hunk 2]
```diff
@@ -28,6 +28,14 @@ type EndpointIPAMConfig struct {
 	LinkLocalIPs []string `json:",omitempty"`
 }
 
+// Copy makes a copy of the endpoint ipam config
+func (cfg *EndpointIPAMConfig) Copy() *EndpointIPAMConfig {
```
Comment (AkihiroSuda) [inline api/types/network/network.go]: WDYT about having `pkg/deepcopy` that implements generic deepcopy via `reflect`? (in another PR)

Comment (cpuguy83) [inline api/types/network/network.go]: I think this is very difficult to do correctly, particularly when non-exported types/fields come into play.

Comment (AkihiroSuda) [inline api/types/network/network.go]: OK

**Comment by cpuguy83** [comment]
Failure is unrelated.

**Comment by cpuguy83** [comment]
And Windows is indeed fixed.

**Review by LK4D4 — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
ping @cpuguy83 @vieux I think @cpuguy83 suggested we should take this one for 1.13.1, please confirm / check

**Comment by cpuguy83** [comment]
Yeah I'm not sure. We have a lot of memory corruption issues it seems in 1.13.0.
So far the only thing I've found that's new is this autremove check, and that couple with the "container.HostConifg = setdefualt(container.HostConfig)" is pretty gross and can be frequently hit depending on the workload.

File: daemon/start.go [hunk 3]
```diff
@@ -138,10 +137,6 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 		return err
 	}
 
-	// Make sure NetworkMode has an acceptable value. We do this to ensure
-	// backwards API compatibility.
-	container.HostConfig = runconfig.SetDefaultNetModeIfBlank(container.HostConfig)
-
 	if err := daemon.initializeNetworking(container); err != nil {
```
Comment (dnephin) [inline daemon/start.go]: This was removed because we already set the default in `create`, so we shouldn't need to again in `start`. Is that right?

**Review by dnephin — APPROVED** [review]
LGTM

File: daemon/start.go (hunk 3 above)
Comment (cpuguy83) [inline daemon/start.go]: Correct.

**Pull request merged**